src/XrdVoms/XrdVomsgsi.cc

```diff
@@ -57,7 +59,11 @@ int XrdSecgsiVOMSFun(XrdSecEntity &ent)
 {
 // Make sure we were initialized. If so, invoke the function and return result.
 //
-   return (vomsFun ? vomsFun->VOMSFun(ent) : -1);
+   int retval = (vomsFun ? vomsFun->VOMSFun(ent) : -1);
+   if (retval == -1) {return retval;}
+
+   auto mapfile = XrdVomsMapfile::Get();
+   return mapfile ? mapfile->Apply(ent) : retval;
```

Comment:
As per my previous comments, we really don't need a change here. The mapping should be part of the core VomsFun and the interface to the xroot-side initialization should pretty much do what the http side does. That way it's all in one place and you clear can see it's part of the core logic. Plus it guarantees that the map will be applied if the "voms.mapfile" directive is in the config file irrespective of the protocol.

Reply:
Resolving conversation - this is now done inside the VOMSFun core.